web/client-api/src/main/java/io/deephaven/web/client/api/parse/JsDataHandler.java

```diff
@@ -94,32 +94,32 @@ public void write(Object[] data, ParseContext context, JsConsumer<Node> addNode,
     },
     DATE_TIME(Type.Int, "io.deephaven.time.DateTime", "datetime") {
         // Ensures that the 'T' separator character is in the date time
-        private String ensureSeparator(String s, ParseContext context) {
-            if (s.charAt(context.separatorIndex) == ' ') {
+        private String ensureSeparator(String s) {
+            if (s.charAt(SEPARATOR_INDEX) == ' ') {
                 StringBuilder stringBuilder = new StringBuilder(s);
-                stringBuilder.setCharAt(context.separatorIndex, 'T');
+                stringBuilder.setCharAt(SEPARATOR_INDEX, 'T');
                 return stringBuilder.toString();
             }
             return s;
         }
 
-        // Updates the pattern for the correct number of subsecond digits 'S'
-        private String getSubsecondPattern(String s, ParseContext context) {
+        // Guess the pattern for the correct number of subsecond digits 'S'
+        private String getSubsecondPattern(String s) {
             final int decimalIndex = s.indexOf('.');
             if (decimalIndex == -1) {
                 // No subsecond digits
-                return context.dateTimePattern;
+                return DEFAULT_DATE_TIME_PATTERN;
```

Comment:
why use this constant instead of the specified pattern in the context (which will default to the same constant)?

Reply:
The pattern in the context can change

Reply:
Yes, isn't that the point? if the user's pattern (which could be the default), and the data changes partway (as we seem to allow differences in data, with additional subsecond precision), wouldnt it make sense to base it on the provided format? Or do we think the user will switch formats entirely when their data changes?

I guess I don't understand the use case where the data changes, and we want to _replace_ the specified format, but not use the original format later?